### PR TITLE
Use deploy-tool in the argo-cd-release action

### PR DESCRIPTION
**Pull request by IuryAlves**
Ticket: https://voidev.atlassian.net/jira/software/projects/FLAT/boards/96?selectedIssue=FLAT-2332

### CODE REVIEW

**Review by jlevesy — APPROVED** [review]
Sounds good, but this is kind of a big bang approach. 

Would that make sense in your case to have it running on a few projects before generalizing it? 

For instance you could create a duplicate `build-and-release-to-env-beta` action that does that, and roll it out on a few candidates projects?

**Comment by IuryAlves** [comment]
> Sounds good, but this is kind of a big bang approach.
> 
> Would that make sense in your case to have it running on a few projects before generalizing it?
> 
> For instance you could create a duplicate `build-and-release-to-env-beta` action that does that, and roll it out on a few candidates projects?

That approach would take time because I would wait for teams to merge my changes in their services.That is the issue with not having versioning in our actions repo. 
I did test these changes with a few services locally though.
@jlevesy

**Comment by jlevesy** [comment]
All good then!

File: argo-cd-release/action.yaml [hunk 1]
```diff
@@ -57,6 +65,23 @@ runs:
       with:
         ssh-private-key: ${{ inputs.bot_ssh_private_key }}
 
+    - name: Authenticate to GCP
+      uses: google-github-actions/auth@v1.0.0
```
Comment (K-Phoen) [inline argo-cd-release/action.yaml:69]:
```suggestion
      uses: google-github-actions/auth@v1
```

Unless there's a reason to pin this to 1.0.0?

File: build-and-release-to-env/action.yaml [hunk 2]
```diff
@@ -90,7 +90,7 @@ runs:
 
     # Authentication steps
     - name: Authenticate to GCP
-      uses: google-github-actions/auth@v1
+      uses: google-github-actions/auth@v0.8.1
```
Comment (K-Phoen) [inline build-and-release-to-env/action.yaml:93]:
```suggestion
      uses: google-github-actions/auth@v1
```

**Review by K-Phoen — APPROVED** [review]
Did you run it, to verify it works on a service?

**Comment by IuryAlves** [comment]
> Did you run it, to verify it works on a service?

Yes, I tested on the `voiury` service. I also tested locally in real services.

**Pull request merged**